cubic-server/Dimension.cpp

```diff
@@ -114,15 +123,15 @@ std::shared_ptr<thread_pool::Task> Dimension::loadOrGenerateChunk(int x, int z,
 
         // This send the chunk to the players that are loading it
         this->_loadingChunksMutex.lock();
-        for (auto &entity : this->_entities) {
+        for (auto &player : this->_players) {
             // pls don't kill me
             // this is a hack to check if the client is still connected
             // And the best part ? I don't even know if it works
-            if (std::find_if(this->_loadingChunks[{x, z}].players.begin(), this->_loadingChunks[{x, z}].players.end(), [entity](const Player *player) {
-                    return player == entity;
+            if (std::find_if(this->_loadingChunks[{x, z}].players.begin(), this->_loadingChunks[{x, z}].players.end(), [player](const std::shared_ptr<Player> current_player) {
```

Comment:
Like written previously, this was a hack to see if the client was still connected, but now we can just get a hold of `weak_ptr` to player or client, try to get a `shared_ptr` from it, and send the chunk accordingly. And if it fails the client is no longer connected.
Pls fix ? :sweat_smile:
I can do it if you want, this is my shitty code anyway